File: backend/app/models/asset_snapshot.py
# backend/app/models/asset_snapshot.py
from datetime import datetime, date
from sqlalchemy import Column, Integer, Date, Numeric, DateTime
from app.database import Base


class AssetSnapshot(Base):
    """일별 자산 스냅샷 - 팀 전체 자산 히스토리 추적용"""
    __tablename__ = "asset_snapshots"

    id = Column(Integer, primary_key=True, index=True)
    snapshot_date = Column(Date, unique=True, index=True, nullable=False)

    # KRW 자산
    krw_cash = Column(Numeric(20, 2), default=0)  # KRW 현금
    krw_evaluation = Column(Numeric(20, 2), default=0)  # KRW 평가액

    # USD 자산
    usd_cash = Column(Numeric(20, 4), default=0)  # USD 현금
    usd_evaluation = Column(Numeric(20, 4), default=0)  # USD 평가액

    # USDT 자산
    usdt_evaluation = Column(Numeric(20, 4), default=0)  # USDT 평가액

    # 합산 (KRW 기준)
    total_krw = Column(Numeric(20, 2), default=0)  # 전체 자산 (KRW 환산)
    exchange_rate = Column(Numeric(10, 2), nullable=True)  # 스냅샷 당시 환율

    created_at = Column(DateTime, default=datetime.utcnow)

File: spring-backend/backend/src/main/java/com/fundmessenger/asset/entity/AssetSnapshot.java
package com.fundmessenger.asset.entity;

import io.hypersistence.utils.hibernate.type.json.JsonType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Type;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;

@Entity
@Table(name = "asset_snapshots")
@Getter
@Setter
@NoArgsConstructor
public class AssetSnapshot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "snapshot_date", unique = true, nullable = false)
    private LocalDate snapshotDate;

    @Column(name = "krw_cash", precision = 20, scale = 2, columnDefinition = "numeric(20,2) default 0")
    private BigDecimal krwCash = BigDecimal.ZERO;

    @Column(name = "krw_evaluation", precision = 20, scale = 2, columnDefinition = "numeric(20,2) default 0")
    private BigDecimal krwEvaluation = BigDecimal.ZERO;

    @Column(name = "usd_cash", precision = 20, scale = 4, columnDefinition = "numeric(20,4) default 0")
    private BigDecimal usdCash = BigDecimal.ZERO;

    @Column(name = "usd_evaluation", precision = 20, scale = 4, columnDefinition = "numeric(20,4) default 0")
    private BigDecimal usdEvaluation = BigDecimal.ZERO;

    @Column(name = "usdt_evaluation", precision = 20, scale = 4, columnDefinition = "numeric(20,4) default 0")
    private BigDecimal usdtEvaluation = BigDecimal.ZERO;

    @Column(name = "total_krw", precision = 20, scale = 2, columnDefinition = "numeric(20,2) default 0")
    private BigDecimal totalKrw = BigDecimal.ZERO;

    @Column(name = "exchange_rate", precision = 10, scale = 2)
    private BigDecimal exchangeRate;

    @Column(name = "realized_pnl", precision = 20, scale = 2, columnDefinition = "numeric(20,2) default 0")
    private BigDecimal realizedPnl = BigDecimal.ZERO;

    @Column(name = "unrealized_pnl", precision = 20, scale = 2, columnDefinition = "numeric(20,2) default 0")
    private BigDecimal unrealizedPnl = BigDecimal.ZERO;

    @Type(JsonType.class)
    @Column(name = "position_details", columnDefinition = "jsonb")
    private Object positionDetails;

    @Column(name = "created_at", columnDefinition = "timestamptz")
    private OffsetDateTime createdAt;
}
